services/motion/motion.go

```diff
@@ -39,6 +40,17 @@ type Service interface {
 		slamName resource.Name,
 		extra map[string]interface{},
 	) (bool, error)
+	MoveOnGlobe(
+		ctx context.Context,
+		componentName resource.Name,
+		destination *geo.Point,
+		heading float64,
```

Comment:
How will we distinguish between not specified and zero here?

Reply:
@edaniels what's the idiomatic way to handle optional number arguments? Should we make this variadic, or create a parameter struct?

Reply:
If it's very infrequently used, I think a struct is okay if it doesn't grow beyond let's say (arbitrarily) 5 members. If it's used a fair bit of the time, math.NaN is a choice. Alternatively, two methods are also acceptable (one with and without heading) if we do not think the combinations will explode over time!

Reply:
I believe this method and this parameter will get used far more as our robots, and customers, come up with more sophisticated ways of using this method, so that sounds like I should lean away from a struct. `math.NaN` is interesting.